core/psh/psh.c

```diff
@@ -972,7 +974,117 @@ static void psh_signalstop(int sig)
 }
 
 
-static int psh_run(void)
+static int psh_login(void)
+{
+	int insz = -1, code;
+	const int maxinsz = 32;
+	char c, indata[maxinsz+1], indatasave[maxinsz+1], shdw[128];
+	struct termios mode;
+
+	tcgetattr(0, &mode);
+	mode.c_lflag |= (ICANON | ECHO);
+	tcsetattr(0, TCSANOW, &mode);
+
+	/* Get login from user */
+	memset(indata, '\0', sizeof(indata));
+
+	for (;;) {
+		if (insz < 0) {
+			fprintf(stdout, "Login: ");
+			fflush(stdout);
+			insz++;
+		}
+
+		if(read(STDIN_FILENO, &c, 1) != 1)
+			continue;
```

Comment:
what will happen if I input `use^D` (CTRL+D -> EOF)

also, missing space after `if`

Reply:
space added. Ctrl+D input:
- does nothing during logging in
- after changes to `psh_readcmd()` properly exits the `psh_run()` closing psh/presenting login prompt (if logging in enabled)